homeassistant/components/sensor/__init__.py

```diff
@@ -311,12 +311,28 @@ def state(self) -> Any:
             # Convert the date object to a standardized state string.
             if device_class == DEVICE_CLASS_DATE:
                 return value.date().isoformat()
+
+            if value.tzinfo is not None and value.tzinfo != timezone.utc:
```

Comment:
Should we do this before converting to a date if the device class is date? I'm thinking the timezone could change the the calculated date.

Reply:
Nice catch!

Adjusted in 8b70af5, added tests for this case as well.